plugins/fabric8-end2end-flow/src/main/java/com/redhat/che/end2end/End2EndFlowService.java

```diff
@@ -89,15 +95,40 @@ public String siteKey() {
   @Path("verify")
   @Consumes("text/plain")
   @Produces(APPLICATION_JSON)
-  public Response verifyReCaptchaToken(String token) {
+  public Response verifyReCaptchaToken(String token, @Context HttpServletRequest servletRequest) {
     int responseCode = 500;
     try {
       URL obj = new URL("https://www.google.com/recaptcha/api/siteverify");
       HttpsURLConnection con = (HttpsURLConnection) obj.openConnection();
 
       con.setRequestMethod("POST");
 
-      String urlParameters = "secret=" + reCaptchaSecretKey + "&response=" + token;
+      String ip = retrieveClientIp(servletRequest);
+
+      String remoteIpPart = "";
+
+      if (ip != null) {
+        LOG.info("Starting reCaptcha verification for user from IP: {}", ip);
```

Comment:
Will this flood the logs once deployed?

Reply:
Well, I don't expect it since token verification will be only called when a user will try to *register* into `che.openshift.io` only, and not on every login.

Reply:
:+1: